src/modules/commander/Commander.cpp

```diff
@@ -123,6 +123,7 @@ static constexpr uint64_t HOTPLUG_SENS_TIMEOUT = 8_s;	/**< wait for hotplug sens
 static constexpr uint64_t PRINT_MODE_REJECT_INTERVAL = 500_ms;
 static constexpr uint64_t INAIR_RESTART_HOLDOFF_INTERVAL = 500_ms;
 
+static constexpr uint8_t AVOIDANCE_MAX_TRIALS = 4;  /* Maximum number of trials for avoidance system to start */
```

Comment:
I don't understand why there are multiple trials. As far as I can see you're just waiting for heartbeats coming from "avoidance", so you just check after the bootup timeout, right? And then you reset the time and check after the bootup timeout again. Aha, I see what's going on, the bootup timeout is not a timeout, it's actually a "check interval".

Reply:
Yes you are right, maybe calling it "check interval" is more appropriate. The reasoning behind this implementation is that we would like to have a message that periodically shows that the system is still booting up, not just wait for a single timeout. Let's say that you have an onboard computer (not `controller` :smile: ) boots in 60s:
- You define the COM_ONB_BOOT_T to 25s
- When you power the system, you will then have a periodic message shows the system is still booting up (as you may now on some companion computers with docker the booting time is still quite high, even more than 2minutes).
- If after three messages (elapsed time will be > than average boot time) the FC has still not received an heartbeat then you alert the user that something is wrong.

Thus printing a message gives the user the impression that the system is actually doing something.


What do you think?
Maybe some parameter renaming could help understanding.

Reply:
What if you only cover these two cases with messages:

Successful case:
(booting, no message, pre-arm check would fail if you try to arm)
When avoidance connects: Send status message "Avoidance system connected" -> Now you know you can fly.

Unsuccessful case:
(booting, no message, pre-arm check would fail if you try to arm)
After timeout (e.g. 2 mins): Send error message "Error: Avoidance system not available!".

This way you set  the timeout that you expect and you have a message that it is ready before that if it's faster than expected.

Reply:
I changed the messaging logic to the one you suggested.
@julianoes  have a look here e004f77